### DISCUSSION TITLE
Are Kuzzle subscriptions/notifications persistent ?

### DISCUSSION THREAD

**Discussion by yhilem** [kuzzleio/kuzzle#2406] [Q&A] [answered]
Hi,
I want to develop an Android Mobile application that subcribe to database events with MQTT protocol : 

- "Each time a subscription is sent, a dedicated MQTT topic is created, named after the channel property issued by Kuzzle." (https://docs.kuzzle.io/core/2/api/protocols/mqtt/#app).
- Notification are meant to be dispatched on "channels" created when subscribing. (https://github.com/kuzzleio/kuzzle/blob/master/lib/core/realtime/notifier.js#L37)

What driver used by aedes for persistence (https://github.com/moscajs/aedes#persistence)?
This parameter is not specified in the Kuzzle configuration.

What happens while my app is disconnected and reconnect ? Are notifications lost ?

Thanks
Youcef HILEM
DSI BSCC La Poste

**Comment by Aschen** [comment, marked as answer]
Hi @yhilem 

Kuzzle notification system is not designed to be persistent by itself. The MQTT protocol is simply used as a transport layer for realtime notification.

There is no persistence option for Aedes because when a client disconnect from Kuzzle, his subscription are also destroyed. 

So to answer your question, when your app disconnect and reconnect, any notification generated during this interval while not be available.

There is few ways to solve this problem:
 - retrieve the current state when application reconnect (with `document:search` for example)
 - persist notifications in a message queue, it could be Redis Stream, Kafka, or even the Aedes broker but all those solutions will involve custom backend code. (the workflow plugin is the easiest way to publish notification in a message queue in a resilient way)
